GitUI/HelperDialogs/FormRemoteProcess.cs

```diff
@@ -26,42 +26,33 @@ public partial class FormRemoteProcess : FormProcess
             new TranslationString("Host Fingerprint not registered");
         #endregion
 
-        public bool Plink { get; set; }
         private bool _restart;
-        protected readonly GitModule Module;
-
-        // only for translation
-        protected FormRemoteProcess()
-        {
-        }
-
-        public FormRemoteProcess(GitModule module, string process, ArgumentString arguments)
-            : base(process, arguments, module.WorkingDir, null, true)
-        {
-            Module = module;
-        }
+        private string _urlTryingToConnect = string.Empty;
 
-        public FormRemoteProcess(GitModule module, ArgumentString arguments)
-            : base(null, arguments, module.WorkingDir, null, true)
+        [Obsolete("For VS designer and translation test only. Do not remove.")]
+        public FormRemoteProcess()
+            : base()
         {
-            Module = module;
+            InitializeComponent();
         }
 
-        public static new bool ShowDialog(GitModuleForm owner, ArgumentString arguments)
+        public FormRemoteProcess([NotNull] GitUICommands commands, string process, ArgumentString arguments)
+            : base(process, arguments, commands?.Module?.WorkingDir, null, true)
         {
-            return ShowDialog(owner, owner.Module, arguments);
+            Commands = commands ?? throw new ArgumentNullException(nameof(commands));
         }
```

Comment:
Consider a factory method for checking the argument `command` without instantiating the `Form`.

Reply:
This is an exceptional path, there are no existing call sites that pass `null`.
I just removed a dozen of "factory" methods.